### ISSUE TITLE
[Animated] Spring doesn't respect overshootClamping as a hard stop

### ISSUE THREAD

**Issue by cmcewen** [facebook/react-native#2409]
Using the Animated library + PanResponder to build a swipeable cell in a ListView with a spring animation. 

I would expect setting overshootClamping to 'true' would make it impossible to move beyond a value I specifically animate to - for example, moving from negative numbers up to 0 should never result in a positive number. However, I can easily swipe the cells in a way where the animation will lead to a positive number, and overshoot the value I specifically animate to.

I could imagine either specifying a range of acceptable values, or modifying the spring animation to make sure that it never reaches a value on the "opposite" side of a value specified.

**Comment by ide** [comment]
cc @sahrens

@cmcwen for your use case you may be able to use the interpolate function, which lets you map an animated value from its domain e.g. [-100, 0] to a range (also [-100, 0] in your case since you don't need to transform the value) and more importantly you can clamp it. Search Animated.js for "clamp" and look at how the UIExplorer demo uses the interpolate function for an example.

overshootClamping should be fixed if it indeed is broken but at least you may be able to work around it for now. Thanks for the report.

**Comment by cmcewen** [comment]
@ide thanks, using the interpolate function seems to have solved that problem.

**Comment by cmcewen** [comment]
Anything I can do on this one? Happy to work on a fix, but it sounds like Animated may be moving towards native execution so this won't be an issue? If so I'll close this.

**Comment by ide** [comment]
cc @vjeux @sahrens ^

**Comment by sahrens** [comment]
We'll most likely always have the JS implementation for cross-platform compatibility, and the way we implement things in native may actually be to run the same JS, but in a different thread in an isolated environment.

tl;dr: feel free to fix and send a PR ;)

**Comment by sahrens** [comment]
My guess is that we just forgot to implement it or broke it when we pulled part of the implementation out of rebound.

**Comment by vjeux** [comment]
> My guess is that we just forgot to implement it or broke it when we pulled part of the implementation out of rebound.

Correct. Pull request to fix that would be great :)

**Comment by mkonicek** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with ProductPains! Check out its new home: https://productpains.com/post/react-native/animated-spring-doesnt-respect-overshootclamping-as-a-hard-stop

ProductPains helps the community prioritize the most important issues thanks to its voting feature.
It is easy to use - just login with GitHub.

Also, if this issue is a bug, please consider sending a PR with a fix. We rely on the community to provide
bugfixes as the core team is heavily focused working on performance.
